packages/compiler/core/messages.ts

```diff
@@ -303,19 +309,32 @@ const diagnostics = {
   "unsupported-default": {
     severity: "error",
     messages: {
-      default: paramMessage`Default values are not supported for '${"type"}' type`,
+      default: paramMessage`Default must be a value type but was a '${"type"}' type`,
```

Comment:
```suggestion
      default: paramMessage`Default must be have a value type but has type'${"type"}'`,
```

I'm not sure if we should call this "value type", though. That has a lot of baggage.